File: internal/types/micro_tbox.go
package types

type MicroTBox struct {
	Classes    []ClassDecl         `json:"classes"`
	Properties []PropertyDecl      `json:"properties"`
	Shapes     []ShapeDecl         `json:"shapes"`
	Aliases    map[string][]string `json:"aliases"`
	Axioms     []FreeAxiom         `json:"axioms"`
	Confidence float64             `json:"confidence"`
}

type ClassDecl struct {
	ID           string   `json:"id"`
	Label        string   `json:"label"`
	SubClassOf   *string  `json:"subClassOf"`
	DisjointWith []string `json:"disjointWith"`
	Evidence     string   `json:"evidence"`
}

type PropertyDecl struct {
	ID              string   `json:"id"`
	Label           string   `json:"label"`
	Domain          string   `json:"domain"`
	Range           string   `json:"range"`
	Characteristics []string `json:"characteristics"`
	InverseOf       *string  `json:"inverseOf"`
	Evidence        string   `json:"evidence"`
}

type ShapeDecl struct {
	TargetClass string            `json:"target_class"`
	Constraints []ShapeConstraint `json:"constraints"`
	Evidence    string            `json:"evidence"`
}

type ShapeConstraint struct {
	Property string   `json:"property"`
	MinCount *int     `json:"min_count"`
	MaxCount *int     `json:"max_count"`
	Datatype *string  `json:"datatype"`
	InValues []string `json:"in_values"`
}

type FreeAxiom struct {
	Statement string `json:"statement"`
	Evidence  string `json:"evidence"`
}

type Triple struct {
	Subject   string `json:"s"`
	Predicate string `json:"p"`
	Object    string `json:"o"`
}

File: ontology-reasoner-net/WeKnora.OntologyReasoner.Core/Models/MicroTBox.cs
using System.Text.Json.Serialization;

namespace WeKnora.OntologyReasoner.Core.Models;

public record MicroTBox
{
    [JsonPropertyName("classes")]
    public List<ClassDecl> Classes { get; init; } = [];

    [JsonPropertyName("properties")]
    public List<PropertyDecl> Properties { get; init; } = [];

    [JsonPropertyName("shapes")]
    public List<ShapeDecl> Shapes { get; init; } = [];

    [JsonPropertyName("aliases")]
    public Dictionary<string, List<string>> Aliases { get; init; } = new();

    [JsonPropertyName("axioms")]
    public List<FreeAxiom> Axioms { get; init; } = [];

    [JsonPropertyName("confidence")]
    public double Confidence { get; init; }
}

public record ClassDecl
{
    [JsonPropertyName("id")]
    public required string Id { get; init; }

    [JsonPropertyName("label")]
    public required string Label { get; init; }

    [JsonPropertyName("subClassOf")]
    public string? SubClassOf { get; init; }

    [JsonPropertyName("disjointWith")]
    public List<string> DisjointWith { get; init; } = [];

    [JsonPropertyName("evidence")]
    public required string Evidence { get; init; }
}

public record PropertyDecl
{
    [JsonPropertyName("id")]
    public required string Id { get; init; }

    [JsonPropertyName("label")]
    public required string Label { get; init; }

    [JsonPropertyName("domain")]
    public required string Domain { get; init; }

    [JsonPropertyName("range")]
    public required string Range { get; init; }

    [JsonPropertyName("characteristics")]
    public List<string> Characteristics { get; init; } = [];

    [JsonPropertyName("inverseOf")]
    public string? InverseOf { get; init; }

    [JsonPropertyName("evidence")]
    public required string Evidence { get; init; }
}

public record ShapeDecl
{
    [JsonPropertyName("target_class")]
    public required string TargetClass { get; init; }

    [JsonPropertyName("constraints")]
    public List<ShapeConstraint> Constraints { get; init; } = [];

    [JsonPropertyName("evidence")]
    public required string Evidence { get; init; }
}

public record ShapeConstraint
{
    [JsonPropertyName("property")]
    public required string Property { get; init; }

    [JsonPropertyName("min_count")]
    public int? MinCount { get; init; }

    [JsonPropertyName("max_count")]
    public int? MaxCount { get; init; }

    [JsonPropertyName("datatype")]
    public string? Datatype { get; init; }

    [JsonPropertyName("in_values")]
    public List<string>? InValues { get; init; }
}

public record FreeAxiom
{
    [JsonPropertyName("statement")]
    public required string Statement { get; init; }

    [JsonPropertyName("evidence")]
    public required string Evidence { get; init; }
}

public record TripleDto
{
    [JsonPropertyName("s")]
    public required string S { get; init; }

    [JsonPropertyName("p")]
    public required string P { get; init; }

    [JsonPropertyName("o")]
    public required string O { get; init; }
}
